### DISCUSSION TITLE
view pdf  side by side

### DISCUSSION THREAD

**Discussion by Eckeyyy** [stephanrauh/ngx-extended-pdf-viewer#2210] [Q&A]
I want use API get the pdf text and x,y location  and generate a new box to view these text  but I can't locate these text exactly

**Comment by stephanrauh** [comment]
I'm afraid I didn't get your question. There seem to be a couple of words missing.

- Which (x,y) coordinates are you referring to? The coordinates of which text?
- Where do you want to put the text box? Should it be part of the PDF file, or do you want to put it outside?
- What's viewing to PDF files side-by-side got to do with putting text in a box?

I apologize if these questions sound stupid. It's just that I don't know what you're trying to do, so I need more context to understand the question.

**Reply by Eckeyyy** [reply]
Hi Stephan,
 Thanks for your relay!
 I did not clear my question  before.
 I want to a feature that we can translate pdf
when the pdf loaded there is  a translate button on toolbar and if I click it 
will render a same pdf by side just like the picture below.
I used the api getPageAsLine  and I can get text of every page 
[{
    "direction": "ltr",
    "x": 256.73,
    "y": -758.76,
    "width": 81.58656000000008,
    "height": 10.559999999999945,
    "text": "ICH M11 Template"
}]
I want to render these text by html but it is hard to located.
![Uploading image.png…]()

**Reply by Eckeyyy** [reply]
![Image](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/136833693/d2d32dd7-541b-40b7-8c77-feaa0dd085f6)

**Reply by stephanrauh** [reply]
Indeed. Have a look at `PDFViewerApplication.pdfViewer._pages[page].pdfPage.view`. That's an array that contains the dimension of the PDF file in pdf.js coordinates. In contrast, `PDFViewerApplication.pdfViewer._pages[page].div.clientWidth`/ `...clientHeight` gives you the size of the PDF page on the screen. You can probably use that to convert pdf.js coordinates to screen coordinates. I've done something similar (just the other way round) here: https://github.com/stephanrauh/ngx-extended-pdf-viewer/blob/2e348520655da23e2cdc2e0ed658168afca2bb05/projects/ngx-extended-pdf-viewer/src/lib/ngx-extended-pdf-viewer.service.ts#L522

Generally speaking, determining the coordinates of a text and rendering it in HTML is a very hard problem. PDF fonts differ from the native fonts of your browser. The pdf.js team has spent a lot of time on it, and they're still not perfect. But they're remarkably close. So you could try to extract the position of the text from the text layer. That's another `<div>` that's optionally rendered on top of the canvas. It's used for seletecting and highlighting text.

**Reply by Eckeyyy** [reply]
ok ,I will try to get it by textlayer ,and one more question ,how can I get the current pagenumber when the pdf is scroll.

**Reply by stephanrauh** [reply]
There's an event listener for that: `(page)`. You can see the live demo on the second tab of https://pdfviewer.net/extended-pdf-viewer/two-way-binding.
